src/NHibernate/Cache/CacheFactory.cs

```diff
@@ -67,16 +67,14 @@ public static class CacheFactory
 						"cache usage attribute should be read-write, read-only, nonstrict-read-write, or transactional");
 			}
 
-			ICache impl;
 			try
 			{
-				impl = settings.CacheProvider.BuildCache(name, properties);
+				ccs.Cache = settings.CacheProvider.BuildCache(name, properties);
```

Comment:
This property setter has no logic, doing the assignment outside of the `try` was just complicating the code for no reasons.